src/gui/MainWindow.cpp

```diff
@@ -6,18 +6,41 @@
 #include "ui_MainWindow.h"
 
 #include <QCheckBox>
+#include <QList>
 #include <QPen>
 
 #include <qwt_legend.h>
 #include <qwt_picker_machine.h>
 
-MainWindow::MainWindow(QWidget* parent)
+#include <cstring>
+
+namespace {
+
+QString format_device_name(QString buffer) {
+    const char* c;
+    if (buffer.toStdString() == "default")
+        return buffer;
+    else {
+        QByteArray temp = buffer.toLocal8Bit();
+        buffer = "hw:";
+        c = (std::strstr(temp.data(), "card ") + 5); // get card number
+        buffer.push_back(*c); // hw:X
+        c = (std::strstr(temp.data(), " device ") + 8); // get device
+        buffer.append(","); // hw:X,
+        buffer.push_back(*c); // hw:X,Y
+        return buffer;
+    }
+}
```

Comment:
Moved the function back into the class.

I think this `format_device_name` function could be avoided if the device list would not be returned as a vector of plain strings but rather as a vector of structs, which already have the channels set during reading of the input/output devices.

But for now this will also work.

Reply:
Agree, also was thinking about structs here. Yep, we can do it later.